pepper-apis/src/main/java/org/broadinstitute/ddp/route/ReceiveDsmNotificationRoute.java

```diff
@@ -124,4 +137,25 @@ private TestResult parseTestResult(Response response, DsmNotificationPayload pay
 
         return testResult;
     }
+
+    private void recordInitialKitSentTime(Handle handle, StudyDto studyDto, User user, KitType kitType, Instant timestamp) {
+        long kitConfigId = handle.attach(KitConfigurationDao.class)
+                .getKitConfigurationDtosByStudyId(studyDto.getId())
+                .stream()
+                .filter(kit -> kit.getKitTypeId() == kitType.getId())
+                .map(KitConfigurationDto::getId)
+                .findFirst()
+                .orElseThrow(() -> new DDPException(String.format(
+                        "Could not find kit configuration for study %s and kit type %s",
+                        studyDto.getGuid(), kitType.getName())));
+
+        var kitScheduleDao = handle.attach(KitScheduleDao.class);
+        var record = kitScheduleDao.findRecord(user.getId(), kitConfigId).orElse(null);
+
+        if (record != null && record.getInitialKitSentTime() == null) {
+            kitScheduleDao.updateRecordInitialKitSentTime(record.getId(), timestamp);
+            LOG.info("Updated initial kit sent time to {} for participant={}, study={}, kitConfigurationId={}",
+                    timestamp, user.getGuid(), studyDto.getGuid(), kitConfigId);
+        }
```

Comment:
Maybe a `LOG.error()` in the else if we get a message here > 1 for a kit that already has a sent time?

Reply:
Actually, we're not tracking the status _per kit_. Pepper doesn't really need to know that a particular kit is sent. It just needs to know that the first kit, whichever it is, is sent so it can schedule the longitudinal kits. The first kit may be a regular kit or it may be an uploaded kit that only DSM knows about. Also, we WILL be getting > 1 `TESTBOSTON_SENT` events since we have longitudinal kits, so I think adding `log.error()` will be a false positive.

Reply:
Ah, got it.  Thanks for helping me understand.

Reply:
In terms of tracking kit status in order to filter out duplicates and such, I agree on that. I had created a ticket for it a while back: DDP-5004.